src/lib.rs

```diff
@@ -50,32 +53,30 @@ fn get_compact_symbol_table_impl(
     binary_data: &[u8],
     debug_data: &[u8],
     breakpad_id: &str,
-) -> Option<compact_symbol_table::CompactSymbolTable> {
+) -> CompactSymbolResult<compact_symbol_table::CompactSymbolTable> {
     let mut reader = Cursor::new(binary_data);
-    if let Ok(hint) = goblin::peek(&mut reader) {
-        match hint {
-            Hint::Elf(_) => {
-                return elf::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::Mach(_) => {
-                return macho::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::MachFat(_) => {
-                let multi_arch = mach::MultiArch::new(binary_data).ok()?;
-                for fat_arch in multi_arch.iter_arches().filter_map(Result::ok) {
-                    let arch_slice = fat_arch.slice(binary_data);
-                    if let Some(table) = macho::get_compact_symbol_table(arch_slice, breakpad_id) {
-                        return Some(table);
-                    }
-                }
-            }
-            Hint::PE => {
-                return pdb::get_compact_symbol_table(debug_data, breakpad_id);
-            }
-            _ => {}
+    match goblin::peek(&mut reader)? {
+        Hint::Elf(_) => {
+            return elf::get_compact_symbol_table(binary_data, breakpad_id);
+        }
+        Hint::Mach(_) => {
+            return macho::get_compact_symbol_table(binary_data, breakpad_id);
+        }
+        Hint::MachFat(_) => {
+            let multi_arch = mach::MultiArch::new(binary_data)?;
+            let fat_arch = multi_arch.iter_arches().filter(|x| x.is_ok()).next().ok_or(CompactSymbolError::NoneValueError)?;
+            let arch_slice = fat_arch?.slice(binary_data);
+            return macho::get_compact_symbol_table(arch_slice, breakpad_id);
```

Comment:
You changed the meaning of this code: Before, `get_compact_symbol_table` was called for every arch, and now it's just called for the first arch. We need the original meaning, for the following purpose:

Some macOS system libraries contain two architectures in their MultiArch file: x86_64 and x86_64h. (The h is for the "Haswell" CPU architecture.) The two architectures have different breakpadIDs. The only way this code knows which of the two to get symbols for is based on the `breakpad_id` parameter. For example, the first arch in the MultiArch container might parse successfully but then error out because of the breakpad ID mismatch. Then we try to parse the second arch, find out that its breakpad ID is the one we were looking for, and successfully use that second arch.